sdk/storage/Azure.Storage.Common/src/Shared/StorageRequestFailedDetailsParser.cs

```diff
@@ -46,7 +47,15 @@ public override bool TryParse(Response response, out ResponseError? error, out I
                             }
                         }
 
-                        error = new ResponseError(errorCode, message);
+                        if (headerName != null && headerName == Constants.HeaderNames.Version)
+                        {
+                            error = new ResponseError(errorCode, "The provided x-ms-version header is not enabled on this storage account.  Please see https://learn.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services for additional information.\n");
```

Comment:
Probably storing this string in the Constants.cs folder may be better. and then that way when you verify the error message is the same in the test, you can just grab the constant string. Just in case you have to change the message in the future, you don't have to do it in both places

Reply:
Fixed.